lightkurve/seismology/core.py

```diff
@@ -276,11 +354,212 @@ def plot_echelle(self, deltanu=None, numax=None,
+                                     height=230,
+                                     direction='rtl',
+                                     show_value=False,
+                                     sizing_mode='fixed',
+                                     name='stretch')
+
+        def stretch_change_callback(attr, old, new):
+            """TPF stretch slider callback."""
+            fig.select('img')[0].glyph.color_mapper.high = new[1]
+            fig.select('img')[0].glyph.color_mapper.low = new[0]
+
+        stretch_slider.on_change('value', stretch_change_callback)
+        return fig, stretch_slider
+
+
+    def interact_echelle(self, notebook_url='localhost:8888', **kwargs):
+        """ Shows an interactive Echelle diagram based on a Seismology object.
+
+        Optional keywords are the same as the `plot_echelle` function.
+
+        Parameters
+        ----------
+        notebook_url: str
+            Location of the Jupyter notebook page (default: “localhost:8888”)
+            When showing Bokeh applications, the Bokeh server must be explicitly
+            configured to allow connections originating from different URLs.
+            This parameter defaults to the standard notebook host and port.
+            If you are running on a different location, you will need to supply this
+            value for the application to display properly. If no protocol is
+            supplied in the URL, e.g. if it is of the form “localhost:8888”,
+            then “http” will be used.
+        deltanu : float
+            Value for the large frequency separation of the seismic mode
+            frequencies in the periodogram. Assumed to have the same units as
+            the frequencies, unless given an Astropy unit.
+            Is assumed to be in the same units as frequency if not given a unit.
+        numax : float
+            Value for the frequency of maximum oscillation. If a numax is
+            passed, a suitable range one FWHM of the mode envelope either side
+            of the will be shown. This is overwritten by custom frequency ranges.
+            Is assumed to be in the same units as frequency if not given a unit.
+        minimum_frequency : float
+            The minimum frequency at which to display the echelle
+            Is assumed to be in the same units as frequency if not given a unit.
+        maximum_frequency : float
+            The maximum frequency at which to display the echelle.
+            Is assumed to be in the same units as frequency if not given a unit.
+        smooth_filter_width : float
+            If given a value, will smooth periodogram used to plot the echelle
+            diagram using the periodogram.smooth(method='boxkernel') method with
+            a filter width of `smooth_filter_width`. This helps visualise the
+            echelle diagram. Is assumed to be in the same units as the
+            periodogram frequency.
+        """
+
+        try:
+            import bokeh
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [redefinition of unused 'bokeh' from line 21](https://app.codacy.com/manual/barentsen/lightkurve/pullRequest?prid=4468966)